### PR TITLE
Update the UI template tests.

### PR DESCRIPTION
**Pull request by kbidarkar**
* Removed the old bz-id in the doc-strings
* Added bz-id to skip `test_remove_template`
* Other miscellaneous fixes.

### CODE REVIEW

File: tests/foreman/ui/test_template.py [hunk 1]
```diff
@@ -58,7 +52,6 @@ def test_negative_create_template_1(self):
                            custom_really=True, template_type=temp_type)
             self.assertIsNotNone(self.template.wait_until_element
                                  (common_locators["name_haserror"]))
-            self.assertIsNone(self.template.search(name))
```
Comment (elyezer) [inline tests/foreman/ui/test_template.py]: The search is broken?

File: tests/foreman/ui/test_template.py [hunk 2]
```diff
@@ -192,16 +171,12 @@ def test_remove_template(self, template_name):
 
         @Assert: Template removed successfully
 
-        @BZ: 1096333
```
Comment (elyezer) [inline tests/foreman/ui/test_template.py]: Should be `@BZ: 1177756`.

File: tests/foreman/ui/test_template.py [hunk 3]
```diff
@@ -192,16 +171,12 @@ def test_remove_template(self, template_name):
...
-
         """
-
-        entities.ConfigTemplate(name=template_name).create()['name']
+        entities.ConfigTemplate(name=template_name).create()
```
Comment (elyezer) [inline tests/foreman/ui/test_template.py]: :+1:

**Comment by elyezer** [comment]
ACK, pending comments.

File: tests/foreman/ui/test_template.py (hunk 1 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_template.py]: For negative tests, it's not needed. The entity is not created, so there is no need to search for it here. checking for the 'common_locators["name_haserror"]' should be fine.

File: tests/foreman/ui/test_template.py [hunk 4]
```diff
@@ -235,15 +207,13 @@ def test_update_template_os(self):
...
-
         """
         name = gen_string("alpha", 6)
         new_name = gen_string("alpha", 6)
         temp_type = 'provision'
-        os_1_name = entities.OperatingSystem().create()['name']
-        os_2_name = entities.OperatingSystem().create()['name']
-        os_list = [os_1_name, os_2_name]
+        os_list = [
+            entities.OperatingSystem().create()['name'] for _ in range(2)
```
Comment (kbidarkar) [inline tests/foreman/ui/test_template.py]: Sure, Have updated all the instances with create_json()

**Comment by kbidarkar** [comment]
Updated as per comments.

**Comment by Ichimonji10** [comment]
Can you rebase this, please? There are four merge commits after this commit.

**Comment by Ichimonji10** [comment]
Karma given.

**Pull request merged**